### ISSUE TITLE
Table Block: default border in editor but not on front end

### ISSUE THREAD

**Issue by mikachan** [WordPress/gutenberg#45065]
### Description

The Table block has a border by default when inserted in the editor, but there is no default border on the front end.

| Editor | Front end |
| ----- | ---------- |
| <img width="685" alt="image" src="https://user-images.githubusercontent.com/1645628/196399558-bd7abee9-a637-4a2d-a628-6e492a5a4d83.png"> | <img width="1229" alt="image" src="https://user-images.githubusercontent.com/1645628/196399680-03494fa3-547f-429d-ba33-22f607d0ad5b.png"> |

This is easily fixed by applying a border using the block settings, however it can be initially confusing that the two views are different by default.

### Step-by-step reproduction instructions

1. Go to a post or page
2. Insert a table block
3. See the borders in the editor
4. View the post/page on the front end, and see that there are no borders applied by default

### Environment info

WP 6.1 (RC1), GB 14.3.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by lwangdu** [comment]
I have tested the table and I am getting same issues. If you add board manually, it works but the table cell padding doesn't work in frontend. Here are screen shots.

![editor](https://user-images.githubusercontent.com/24829573/196460177-85e83325-189b-460f-9c77-ea081df6c999.png)

![frontend](https://user-images.githubusercontent.com/24829573/196460312-2b5a4844-fe21-4074-8dbb-da3549e3d0af.png)

**Comment by mikachan** [comment]
Thanks @lwangdu, that's similar to what's reported here: https://core.trac.wordpress.org/ticket/56818

I'm not able to replicate this exactly, I don't seem to be able to set any padding on the table cells at all:

<img width="1067" alt="image" src="https://user-images.githubusercontent.com/1645628/196471779-656c12f7-92ca-4676-853a-46a868870cf4.png">

The padding is applied to the table itself, rather than the table cells.

**Comment by mikachan** [comment]
Ah, if I test with GB deactivated and WP 6.1 RC1, I see the table cell padding:

<img width="1069" alt="image" src="https://user-images.githubusercontent.com/1645628/196472654-33322ff5-7fdf-4723-bc05-bce81157c074.png">

Although, I'm still not able to set the sizing with the editor. It looks like the editor used to provide a default padding of `0.5em` to the table cells, but perhaps this was only applied to the editor and not the front end.

Edit:

I think I've tracked this down to the changes in https://github.com/WordPress/gutenberg/pull/44640, where we removed the theme.css from loading in the editor if the theme does not opt into block styles. I believe this is the correct behaviour for now.

To improve this specifically for the table block, perhaps we could always include that default padding value on both the front end and in the editor since it's not currently editable via Global Styles? I've tried this out [here](https://github.com/WordPress/gutenberg/pull/45069).
